src/api/forms/service/migration-helpers.js

```
@@ -71,6 +75,91 @@ export function repositionSummary(definition) {
   return definition
 }
 
+/**
+ * Applies page titles if they are missing
+ * @param {FormDefinition} definition
+ */
+export function applyPageTitles(definition) {
+  const changedPages = definition.pages.map((page) => {
+    if (
+      page.controller !== ControllerType.Summary &&
+      (!page.title || page.title === '')
+    ) {
+      return {
+        ...page,
+        title: hasComponents(page) ? page.components[0].title : ''
+      }
+    }
+
+    return page
+  })
+
+  return {
+    ...definition,
+    pages: changedPages
+  }
+}
+
+/**
+ * Migrates component fields
+ * @param {FormDefinition} definition
+ */
+export function migrateComponentFields(definition) {
+  const changedPages = definition.pages.map((page) => {
+    if (!hasComponentsEvenIfNoNext(page)) {
+      return page
+    }
+
+    const changeComponents = page.components.map((comp) => {
+      if (hasFormField(comp)) {
+        return {
+          ...comp,
+          shortDescription: comp.title
+        }
+      }
+      return comp
+    })
+
+    return {
+      ...page,
+      components: changeComponents
+    }
+  })
+
+  return {
+    ...definition,
+    pages: changedPages
+  }
+}
+
+/**
+ * Converts declaration text to a guidance component
+ * @param {FormDefinition} originalDefinition
+ */
+export function convertDeclaration(originalDefinition) {
+  const definition = structuredClone(originalDefinition)
+
+  const summaryPage = definition.pages.find(
+    (p) => p.controller === ControllerType.Summary
+  )
+  if (!summaryPage) {
+    return definition
```

maybe just throw an error here, since it can't be migrated? if `originalDefinition.declaration` exists and we don't have a summary page, we're out of luck.

---

just to clarify, it'd be something like:

```
if (definition.declaration && !summaryPage) {
    throw new Error('Cannot migrate as unable to find Summary Page')
} else if (!definition.declaration)
   return definition
}

// now do your migration
```